### ISSUE TITLE
Cannot mount or format microSD card

### ISSUE THREAD

**Issue by graysonhead** [Jovian-Experiments/Jovian-NixOS#321]
Hey there. First of all thanks for this project! This is pretty slick, and I'm hoping to get it working well enough that I can use this in favor of the regular OS. But one major issue that is blocking me is that I can't seem to use microSD cards. Whenever I try to format it, it says "We encountered an error while formatting your microSD card (2). Please try again or contact steam-support."

This microSD card also has a pre-existing steam library on it. When I switch to desktop mode on the Jovian NixOS install, I can see all the games in the desktop version of Steam. So I'm wondering perhaps if the microSD card isn't being auto-mounted in the Steam UI session? Anyone run into this before?

**Comment by jptrzy** [comment]
Yes, unfortunately there is no auto mounting of microSD card in the Steam UI
or at list to me knowledge.

*I have the same issue*

**Comment by jptrzy** [comment]
## This is my current walk-around


**hardware-configuration.nix**
``` nix
  fileSystems."/mnt/MicroSDCard" = {
    device = "/dev/mmcblk0p1";
    # If you have this partition mounted, then you can check its type by using
    # df -T | grep /dev/${device}
    fsType = "ext4";
    options = [
      # System will boot up if you don't have sd card inserted
      "nofail"
      # After booting up systemd will try mounting the sd card
      "x-systemd.automount"
    ];
  };
```

This will try mounting /dev/mmcblk0p1 partition at a boot time.
If it is not detected, then the system will boot without any issue, but Systemd will still try to detect if MicroSD Card is inserted and will auto mount it.

*It lacks flexibility to be implemented into main branch,
but it should be ok until someone fix this issue.*

**Comment by Majiir** [comment]
From `.steam/steam/logs/systemmanager.txt`, when clicking Format Storage:

```
[2024-09-08 17:47:58] CClientJobFormatRemovableStorage: starting
[2024-09-08 17:47:58] CClientJobFormatRemovableStorage: stage: 1 progress 0.000000 rtCompletionTime: 0
[2024-09-08 17:47:58] format-sdcard: [JOVIAN] /nix/store/i4zdqksakq68dhvrma9lsh7jda0fhmqb-jovian-stubs/bin/sudo: stub called with: -n /usr/lib/hwsupport/format-sdcard.sh
[2024-09-08 17:47:58] format-sdcard: /nix/store/i4zdqksakq68dhvrma9lsh7jda0fhmqb-jovian-stubs/bin/sudo: line 18: /usr/lib/hwsupport/format-sdcard.sh: No such file or directory
[2024-09-08 17:47:58] format-sdcard returned: 127
[2024-09-08 17:47:59] CClientJobFormatRemovableStorage: result 2
```

**Comment by samueldr** [comment]
We now have mounting:

 - https://github.com/Jovian-Experiments/Jovian-NixOS/pull/413

And formatting is coming:

 - https://github.com/Jovian-Experiments/Jovian-NixOS/pull/418

**Comment by ParetoOptimalDev** [comment]
Maybe I should create a different issue or discussion (let me know), but I've found that even though my ext formatted sd-card is mounting steam itself isn't detecting it.

Maybe there is a specific way I need to format it beyond "click ext4 in gparted and apply operations"?

I am also on steam beta, maybe I should go back to stable and test?

**Comment by K900** [comment]
You need to create a Steam library on it if you've formatted it yourself and not through Steam. You can do that in Steam's desktop mode.
